test/parallel/test-buffer-indexof.js

```diff
@@ -401,13 +401,26 @@ assert.equal(b.lastIndexOf(0x61, Infinity), 0);
 assert.equal(b.lastIndexOf(0x0), -1);
 
 // Test weird offset arguments.
-// Behaviour should match String.lastIndexOf:
-assert.equal(b.lastIndexOf('b', 0), -1);
+// Behavior should match String.lastIndexOf()
+var s = 'abcdef';
+
+// The following offsets coerce to NaN, searching the whole Buffer (or String)
 assert.equal(b.lastIndexOf('b', undefined), 1);
-assert.equal(b.lastIndexOf('b', null), -1);
+assert.equal(s.lastIndexOf('b', undefined), 1);
```

Comment:
I realize there's a lot of `assert.equal()` in this test file, but if you could at least use `assert.strictEqual()` for the stuff you're adding, that would probably be better. If you want to change surrounding instances in the file too, awesome.